trace/kafka/kafka.go

```diff
@@ -5,27 +5,30 @@ import (
 
 	"github.com/Shopify/sarama"
 	"github.com/beatlabs/patron/correlation"
+	"github.com/beatlabs/patron/encoding"
 	"github.com/beatlabs/patron/encoding/json"
 	"github.com/beatlabs/patron/errors"
 	"github.com/beatlabs/patron/trace"
 	"github.com/opentracing/opentracing-go"
 	"github.com/opentracing/opentracing-go/ext"
+	"reflect"
+	"runtime"
 )
 
 // Message abstraction of a Kafka message.
 type Message struct {
 	topic string
-	body  []byte
+	body  interface{}
```

Comment:
We should introduce a MessageRaw and a Message struct to handle the differences... type assertion/reflection is magic and should be avoided...
Go provers:
Clear is better than clever.
Reflection is never clear.

Reply:
I personally prefer that approach as well, but the thing is that we'll have to also create a `NewMessageRaw`, `NewMessageRawWithKey`, `createProducerMessageRaw` etcetera and lead to duplication of much code. 

Are you okay with this?

Reply:
I prefer this over reflection.

Reply:
We could decide to do the following:
Since we will have default JSON encoding, if the user does not provide and encoder, we can remove the Raw totally. This would make everything simple.
Agree?

Reply:
So do we keep `body interface{}`, provide the JSON encoding by default and remove the `SendRaw()` and `rawEncodeFunc()`?

This way
- If someone provides a `[]byte` body, and no encoder, the default (using json Encoder) will return [a base64-encoded string](https://golang.org/pkg/encoding/json/#Marshal).       
**Breaking change** because right now clients are responsible for serializing their messages, and the 'encoder' is just a pass-through.
- If someone provides a `channel/func/mutex` body and no encoder, it will return an error.


This would mean that the user will be responsible for providing the correct Message body type, to match their encodeFunc.

Reply:
We are altering the producer from a thing that gets []byte to something that takes interface{} and encodes it... 
This is breaking behaviour by itself... we have to be really explicit about it...
The alternative would be having all the `Raw` stuff (message, functions).

Reply:
I've implemented this change, let me know if it looks cleaner.